### PR TITLE
Fix profile resizer to make first character of a line selectable in IRC layout

### PR DESCRIPTION
**Pull request by luixxiul**
This PR intends to fix the issue that the first character of a line in IRC layout cannot be selected by replacing the hard-coded inline inset value of `mx_ProfileResizer` with a variable. The issue may be specific to the desktop app for Linux distros (see: https://github.com/vector-im/element-web/issues/14764#issuecomment-735418888; the issue is reproduced on Debian for me). 

To improve accessibility, it is possible to move the resizer a little bit (~5px?) to the left side by setting `- var(--resizer-a11y)` with `--resizer-a11y: 5px`. Let me know if such spacing should be added.

Fixes https://github.com/vector-im/element-web/issues/14764

|Before|After|
|---------|------|
|The resizer overlaps the first character etc.<br />Screenshot from 2023-03-17 04-22-50|Screenshot from 2023-03-17 04-22-07|

type: defect

Signed-off-by: Suguru Hirahara <luixxiul@users.noreply.github.com>

## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Fix profile resizer to make first character of a line selectable in IRC layout ([\#10396](https://github.com/matrix-org/matrix-react-sdk/pull/10396)). Fixes vector-im/element-web#14764. Contributed by @luixxiul.

### CODE REVIEW

**Comment by luixxiul** [comment]
In fact, moving just 3px feels like great a11y improvement for me. You could test it locally by setting `left: calc(70px + var(--name-width))` via browser's devtool.

File: cypress/e2e/timeline/timeline.spec.ts [hunk 1]
```diff
@@ -161,8 +161,22 @@ describe("Timeline", () => {
...
+                // See: _IRCLayout
+                // --RoomView_MessageList-padding = 18px (See: _RoomView.pcss)
+                // --MessageTimestamp-width = $MessageTimestamp_width = 46px (See: _common.pcss)
+                // --icon-width = 14px
+                // --right-padding = 5px
+                // --name-width = 80px
+                // --resizer-width = 15px
+                // 18px + 46px + 14px + 5px + 80px + 5px - 15px
+                // = 153px
+                cy.get(".mx_ProfileResizer").should("have.css", "inset-inline-start", "153px");
```
Comment (luixxiul) [inline cypress/e2e/timeline/timeline.spec.ts:178]: Since the resizer is transparent, it is not possible to use a screenshot test to detect a regression.

**Comment by luixxiul** [comment]
@florianduros Would you please review when you have some time? Thanks in advance.

**Review by florianduros — CHANGES_REQUESTED** [review]

**Comment by florianduros** [comment]
Also I tried to put `left: calc(70px + var(--name-width))` to the `mx_ProfileResizer ` but I didn't see any visible differences. I'am missing something ?

File: res/css/views/rooms/_IRCLayout.pcss [hunk 2]
```diff
@@ -217,11 +217,21 @@ $irc-line-height: $font-18px;
...
-        width: 15px;
-        left: calc(80px + var(--name-width));
+        width: var(--resizer-width);
         cursor: col-resize;
         z-index: 100;
+
+        /* Add width of every element rendered before the resizer (including padding for the avatar and the display
+           name), subtracting the resizer width itself to prevent the resizer from overlapping the text.
+           Please note that MessageTimestamp does not have inline padding. */
+        /* stylelint-disable-next-line declaration-colon-space-after */
```
[the remark this answers is not in the document]
Comment (luixxiul) [reply res/css/views/rooms/_IRCLayout.pcss]: I was too lazy to notice the deprecation messages on the console.

**Review by florianduros — COMMENTED** [review]
Better without the ignore rule, thanks!

About the extra `3px`, I think that's a good idea to improve the usability.  Do you to do it in this PR (which is improving the use case of your fix) ? 

If you want to keep the PR in this current state, I'm happy to merge it.

**Comment by luixxiul** [comment]
> About the extra 3px, I think that's a good idea to improve the usability. Do you want to do it in this PR (which is improving the use case of your fix) ?

Yes, I'm adding the 3px with this PR. Please wait for a moment..

**Review by florianduros — APPROVED** [review]
Great, thank for your contribution !

**Pull request merged**